internal/proxy/clientgo_cached_proxy_resolver.go

```diff
@@ -155,6 +155,17 @@ func (p *clientgoCachedProxyResolver) ObjectExists(obj client.Object) (bool, err
 	return exists, err
 }
 
+func (p *clientgoCachedProxyResolver) IsReady() bool {
```

Comment:
```suggestion
func (p *clientgoCachedProxyResolver) IsKongReady() bool {
or
func (p *clientgoCachedProxyResolver) IsProxyReady() bool {
```

Reply:
It's part of the Proxy interface. IMO the latter option is implicit as such, and methods shouldn't re-state their interface.

Reply:
After more thought, here we are trying to query "is controller ready ?  need ensure able to communicate to Kong before announce KIC is ready"